From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Attard <samuel.r.attard@gmail.com>
Date: Thu, 17 Sep 2020 14:18:30 -0700
Subject: fix: add v8_enable_reverse_jsargs defines in common.gypi

This can be removed once node upgrades V8 and inevitably has to do this exact same thing.  Also hi node people if you are looking at this.

diff --git a/common.gypi b/common.gypi
index e58cc4b4fb4ba28f23133530dc3d908c4cb68426..4fd15f8e80ba82f69aa651a78dd4d567b5e34c5d 100644
--- a/common.gypi
+++ b/common.gypi
@@ -78,6 +78,8 @@
     # TODO(refack): make v8-perfetto happen
     'v8_use_perfetto': 0,
 
+    'v8_enable_reverse_jsargs%': 1,
+
     ##### end V8 defaults #####
 
     # When building native modules using 'npm install' with the system npm,
@@ -399,6 +401,9 @@
       ['v8_enable_pointer_compression == 1 or v8_enable_31bit_smis_on_64bit_arch == 1', {
         'defines': ['V8_31BIT_SMIS_ON_64BIT_ARCH'],
       }],
+      ['v8_enable_reverse_jsargs == 1', {
+        'defines': ['V8_REVERSE_JSARGS'],
+      }],
       ['OS == "win"', {
         'defines': [
           'WIN32',
